### PR TITLE
Allow multiple file groups of any type

### PR DESCRIPTION
**Pull request by sebastian-meyer**
This pull request adds the possibility to specify more than one file group for fulltext, thumbs, audio and downloads. The comma-separated values are checked from left to right and the first fileGrp available in the METS file is used.

This is used in the Zeitungsportal in order to provide a locally hosted fallback for files (`fileGrpFulltext = DDB_FULLTEXT,FULLTEXT`).

### CODE REVIEW

File: Classes/Plugin/AudioPlayer.php [hunk 1]
```diff
@@ -123,18 +129,24 @@ public function main($content, $conf)
...
-            // Get audio data.
-            $this->audio['url'] = $this->doc->getFileLocation($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$this->conf['fileGrpAudio']]);
-            $this->audio['label'] = $this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['label'];
-            $this->audio['mimetype'] = $this->doc->getFileMimeType($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$this->conf['fileGrpAudio']]);
+        $fileGrpAudios = GeneralUtility::trimExplode(',', $this->conf['fileGrpAudio']);
+        while ($fileGrpAudio = array_shift($fileGrpAudios)) {
+            if (!empty($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$fileGrpAudio])) {
+                // Get audio data.
+                $this->audio['url'] = $this->doc->getFileLocation($this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$fileGrpAudio);
```
Comment (albig) [inline Classes/Plugin/AudioPlayer.php]: I think Scrutinizer is right here. The brackets are not correct.

**Review by albig — CHANGES_REQUESTED** [review]
Good idea. Haven't tried it yet. Just stumbled about the Scrutinizer warning.

File: Classes/Plugin/AudioPlayer.php (hunk 1 above)
Comment (sebastian-meyer) [reply Classes/Plugin/AudioPlayer.php]: Fixed with 40c738f.

**Review by beatrycze-volk — COMMENTED** [review]
I wanted to test it but get error:

```
undefined:1 GET https://sdvtypo3ddbzeitungsportal.slub-dresden.de/undefined 404

  | Image (async) |   |  
-- | -- | -- | --
  | dlfUtils.fetchStaticImageData | @ | Utility.js:288
  | (anonymous) | @ | Utility.js:248
  | dlfUtils.fetchImageData | @ | Utility.js:227
  | dlfViewer.initLayer | @ | PageView.js:482
  | dlfViewer.init | @ | PageView.js:376
  | dlfViewer | @ | PageView.js:14
```


I have checked it locally and on test system. In both cases the same error.

**Comment by sebastian-meyer** [comment]
Since `fileGrps` was renamed to `fileGrpImages`, the configuration for the image file groups has to be reloaded. Just open the extension configuration, switch to the "Files" tab, make sure that `file.fileGrpImages` is set correctly and save the configuration.

**Comment by sebastian-meyer** [comment]
I'll try to add a fallback for existing installations to ease updating...

**Review by beatrycze-volk — APPROVED** [review]

**Comment by sebastian-meyer** [comment]
@albig This is ready for merging as soon as you can confirm that the changes you requested (regarding the Scrutinizer warnings) were made.

**Review by albig — APPROVED** [review]
Works as expected.

**Pull request merged**